packages/apollo-engine-reporting/src/extension.ts

```diff
@@ -373,6 +378,12 @@ export const plugin = <TContext>(
         },
 
         didEncounterErrors({ errors }) {
+          // We don't report some special-cased errors to Graph Manager.
+          // See the definition of this function for the reasons.
+          if (allUnreportableSpecialCasedErrors(errors)) {
```

Comment:
If we moved preflight to a didResolveQuerySource handler, this logic would just become "if we didn't get to didResolveQuerySource yet, ignore the errors".

Reply:
I had temporarily thought that the new behavior I'm describing via [this commit](6d4777da1f4dfaab3bf23523e9106e6c2abbab08) (which I've since reverted) would be welcome, but forgot that I'd intentionally maintained parity before.

Instead, I've done what you've suggested here — now enabled by `didResolveSource`'s introduction in https://github.com/apollographql/apollo-server/pull/4076 — which was roughly just reverting what I introduced here in https://github.com/apollographql/apollo-server/commit/78a4cb77edc7c7f11b68f5969a6e85b75e75b6e7.  Overall, still a net reduction in code!

Resolved via: 59b401330064551b20339417ae26fde37ffe3857.